cache-config/t3c/t3c.go

```diff
@@ -64,20 +65,19 @@ func main() {
 	}
 
 	app := "t3c-" + cmd
-	args := append([]string{app}, os.Args[2:]...)
 
-	ex, err := os.Executable()
+	appPath, err := exec.LookPath(app)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "error getting application information: "+err.Error()+"\n")
-		os.Exit(ExitCodeExeErr)
+		fmt.Fprintf(os.Stderr, "error finding path to '"+app+"': "+err.Error())
+		os.Exit(ExitCodeCommandLookupErr)
 	}
-	dir := filepath.Dir(ex)
-	appDir := filepath.Join(dir, app) // TODO use path, not exact dir of this exe
+
+	args := append([]string{app}, os.Args[2:]...)
 
 	env := os.Environ()
 
-	if err := syscall.Exec(appDir, args, env); err != nil {
-		fmt.Fprintf(os.Stderr, "error executing sub-command: "+err.Error()+"\n")
+	if err := syscall.Exec(appPath, args, env); err != nil {
```

Comment:
This warrants some explanation: this was a sort-of bug I found while working on this.
So `syscall.Exec` is calling `execve`, which takes the full path of the executable, _not_ just the name. Just the name appears to work on some *NIX but not others (specifically, it seems to work on OS X, but not CentOS).

So, what we were doing, was getting the path of the current exe (t3c) and prepending that to the app name (e.g. t3c-apply).

This worked as long as they were in the same directory. Which they are with the provided RPM. But we shouldn't rely on that, apps calling other apps should always use the PATH. Specifically, if you're developing and have apps in different build dirs, this then broke things like `t3c apply`. Most app execution methods do this for you, but `execve` doesn't. (And we very specifically want execve not anything else, because it fully replaces the running app with the called one, rather that a subshell or sub-process or anything else messy).

TLDR this is the right solution: we lookup the executable's path from the OS, and then call `execve` with that full path, which fully replaces the running `t3c` with the subcommand like `t3c-apply`.